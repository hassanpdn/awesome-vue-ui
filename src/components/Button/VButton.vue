<template>
      <button class="
            p-3 
            rounded-sm 
            hover:bg-green-50 
            hover:shadow-md 
            hover:shadow-gray-400 
            active:shadow-sm 
            active:bg-green-800 
            transition-colors 
            transition-shadows 
            duration-500
            ease-linear"
            :class="`${expanded ? 'w-full' : ''} ${bgColor} ${textColor}`"
            @click="handleClick"
      >
            <slot/>
      </button>
</template>

<script>
      export default {
            name: 'v-button',
            props: {
                  expanded: {
                        type: Boolean,
                        default: false
                  },
                  textColor: {
                        type: String,
                        default: 'bg-gray-300'
                  },
                  bgColor: {
                        type: String,
                        default: "bg-blue-500",
                  }
            },
            methods: {
                  handleClick(){
                        this.$emit('click')
                  }
            }
      }
</script>

<style>
</style>