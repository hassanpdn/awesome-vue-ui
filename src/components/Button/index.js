import VButton from "./VButton";

export default { VButton }