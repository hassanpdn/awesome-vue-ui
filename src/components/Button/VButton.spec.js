import { mount } from '@vue/test-utils'
import Button from '@/components/Button/VButton.vue';
import { describe, it, expect } from "vitest";

describe('Button', () => {
      const wrapper = mount(Button)
      it('renders a vue instance', () => {
            expect(wrapper.html()).toMatchSnapshot()
      })

      it('renders a button', () => {
            expect(wrapper.find('button').exists()).toBe(true)
      })

      it('should call click handler', async () => {
            await wrapper.find('button').trigger('click')
            expect(wrapper.emitted().click).toBeTruthy()
      })

      it('checks the prop to be passed', () => {
            const wrapper = mount(Button, {
                  propsData: {
                        expanded: true,
                        bgColor: 'bg-red-300'
                  }
            })
            expect(wrapper.classes()).toContain('w-full');
            expect(wrapper.classes()).toContain('bg-red-300');
      })

})