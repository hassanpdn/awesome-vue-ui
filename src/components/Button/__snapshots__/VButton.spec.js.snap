// Vitest Snapshot v1

exports[`Button > renders a vue instance 1`] = `"<button class=\\"p-3 rounded-sm hover:bg-green-50 hover:shadow-md hover:shadow-gray-400 active:shadow-sm active:bg-green-800 transition-colors transition-shadows duration-500 ease-linear bg-blue-500 bg-gray-300\\"></button>"`;
